micrometer-observation/src/main/java/io/micrometer/observation/Observation.java

```diff
@@ -742,12 +742,21 @@ public void setParentObservation(@Nullable ObservationView parentObservation) {
         }
 
         /**
-         * Optional error that occurred while processing the {@link Observation}.
+         * Optional of an error that occurred while processing the {@link Observation}.
          * @return optional error
          */
         @Override
-        public Optional<Throwable> getError() {
-            return Optional.ofNullable(this.error);
+        public Optional<Throwable> getErrorOptional() {
+            return Optional.ofNullable(this.getError());
```

Comment:
Maybe it is my personal preference, but if all we're doing is wrapping the other method with `ofNullable`, I feel like we may as well ask users to do that where they want to use an `Optional` rather than have both ways to maybe get the error. I would personally just have `Throwable getError()`. I don't think we have an optional variant anywhere else in our API. What is the reason we think this one method warrants an optional variant?

Reply:
I might be biased (I like `Optional` a lot) but to me returning an optional here feels right because:
- It's a nice way to indicate the absence of a value which should be the common case here
- It tells the user very explicitly about the intention and forces them to handle both cases (presence/absence of the value)
- Reduces the chance or errors related to null
- Increases readability (I think), see:
```
context.getErrorOptional()
    .map(error -> error.getClass().getSimpleName())
    .orElse("none");
```

On the other hand, since this creates a new instance, it's less performant, that's the justification for `getError`.

Reply:
What I meant to highlight is that none of the the other API in this repo or this module or the Context class returns Optional, so unless there is a specific reason that makes getError different from other places null may be returned (e.g. getContextualName, getParentObservation), it feels inconsistent to have an Optional variant here and not other places. For the sake of consistency, shouldn't we do one of the following: add Optional variants everywhere, not have any Optional variants, or have a reason why getError warrants an Optional variant but other places do not.

I agree it can be a nicer API, but I think the primary use of getError (and other getters on Context) will be in instrumentation on the hot path where we want the least overhead, which means the performance characteristics are especially important, and I think that's the main reason we have avoided Optional in other APIs. If a user wants to use `Optional` it's one `Optional.ofNullable` wrap away, which is all we are doing anyway. And with the nullability annotations we are using, their IDE should let them know the return value may be null.

Reply:
I agree that maybe we should just leave `getError` return `null` and remove the `Optional` version whatsover

Reply:
👍🏼 I removed the Optional variant.